spec/controllers/application_controller_spec.rb

```diff
@@ -265,6 +265,17 @@
     end
   end
 
+  describe "#require_glowfic_domain" do
+    it "redirects on valid requests" do
+      ENV['DOMAIN_NAME'] ||= 'domaintest.host'
+      expect(controller.request.method).to eq('GET')
+      expect(controller.request.xhr?).to be_nil
+      expect(controller.request.host).not_to include('glowfic')
+      expect(controller).to receive(:redirect_to).with("https://domaintest.host", status: :moved_permanently)
+      controller.send(:check_domain)
+    end
```

Comment:
I had a suggestion in the Discord around this:

> for the check_domain test, it might be worthwhile trying to fake the `Host` parameter of the request (<https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#Request_fields>)
> which you should be able to do by specifying `headers: {Host: '…'}` on the get thing: <https://api.rubyonrails.org/v5.2.1/classes/ActionDispatch/Integration/Session.html#method-i-process>
```

That way we can check it happens through a proper flow when we load the site (e.g. by putting it on whatever serves the index page) on the wrong URL, and doesn't happen when we do it on the right URL?

It seems worthwhile to also have this test, though I think it'd make more sense to be labeled `describe "#check_domain"` instead?